### PR TITLE
Force redistribute single-node left when distributed joins are enabled

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

**Comment by dain** [comment]
Before:
```
presto:tiny> EXPLAIN (type distributed)
          -> SELECT a.custkey
          -> FROM (
          ->     SELECT *
          ->     FROM orders
          ->     limit 100
          -> ) a join orders using (custkey);
                                         Query Plan                                          
---------------------------------------------------------------------------------------------
 Fragment 0 [SINGLE]                                                                         
     Output layout: [custkey]                                                                
     Output partitioning: SINGLE []                                                          
     - Output[custkey] => [custkey:bigint]                                                   
         - Project => [custkey:bigint]                                                       
             - InnerJoin[("custkey" = "custkey_52")] => [custkey:bigint, custkey_52:bigint]  
                 - Limit[100] => [custkey:bigint]                                            
                     - RemoteSource[1] => [custkey:bigint]                                   
                 - RemoteSource[2] => [custkey_52:bigint]                                    
                                                                                             
 Fragment 1 [SOURCE]                                                                         
     Output layout: [custkey]                                                                
     Output partitioning: SINGLE []                                                          
     - Limit[100] => [custkey:bigint]                                                        
         - TableScan[tpch:tpch:orders:sf0.01, originalConstraint = true] => [custkey:bigint] 
                 custkey := tpch:custkey                                                     
                                                                                             
 Fragment 2 [SOURCE]                                                                         
     Output layout: [custkey_37]                                                             
     Output partitioning: SINGLE []                                                          
     - TableScan[tpch:tpch:orders:sf0.01, originalConstraint = true] => [custkey_37:bigint]  
             custkey_37 := tpch:custkey                                                      
```
After:
```
                                         Query Plan                                          
---------------------------------------------------------------------------------------------
 Fragment 0 [SINGLE]                                                                         
     Output layout: [custkey]                                                                
     Output partitioning: SINGLE []                                                          
     - Output[custkey] => [custkey:bigint]                                                   
         - RemoteSource[1] => [custkey:bigint]                                               
                                                                                             
 Fragment 1 [HASH]                                                                           
     Output layout: [custkey]                                                                
     Output partitioning: SINGLE []                                                          
     - Project => [custkey:bigint]                                                           
         - InnerJoin[("custkey" = "custkey_37")] => [custkey:bigint, custkey_37:bigint]      
             - RemoteSource[2] => [custkey:bigint]                                           
             - RemoteSource[4] => [custkey_37:bigint]                                        
                                                                                             
 Fragment 2 [SINGLE]                                                                         
     Output layout: [custkey]                                                                
     Output partitioning: HASH [custkey]                                                     
     - Limit[100] => [custkey:bigint]                                                        
         - RemoteSource[3] => [custkey:bigint]                                               
                                                                                             
 Fragment 3 [SOURCE]                                                                         
     Output layout: [custkey]                                                                
     Output partitioning: SINGLE []                                                          
     - Limit[100] => [custkey:bigint]                                                        
         - TableScan[tpch:tpch:orders:sf0.01, originalConstraint = true] => [custkey:bigint] 
                 custkey := tpch:custkey                                                     
                                                                                             
 Fragment 4 [SOURCE]                                                                         
     Output layout: [custkey_37]                                                             
     Output partitioning: HASH [custkey_37]                                                  
     - TableScan[tpch:tpch:orders:sf0.01, originalConstraint = true] => [custkey_37:bigint]  
             custkey_37 := tpch:custkey                                                      
```
Fixed #4610

**Comment by martint** [comment]
looks good

File: presto-docs/src/main/sphinx/release/release-0.139.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+=============
+Release 0.139
+=============
+
+General Changes
+---------------
+
+* Fix planning bug that causes some JOINs to not be redistributed when ``distributed-joins-enabled`` is true.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.139.rst]: Lowercase "joins"

Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.139.rst]: Wrap this line

**Pull request closed without merging**